### PR TITLE
Soften stance on containerization

### PR DESCRIPTION
**Pull request by jnmoyne**

### CODE REVIEW

File: running-a-nats-service/environment.md [hunk 1]
```diff
@@ -11,24 +11,20 @@ If you do use load balancers you just need to understand the potential issues wi
...
 
-NATS is 'cloud native' and expected to be deployed in virtual environments. However, when it comes to maximum performance, such as when running benchmarks in order to estimate the size of future production deployments, it is good to keep a few things in mind.
+NATS is 'cloud native' and expected to be deployed in virtual environments and/or containers.
 
-Think of Core NATS servers are a software equivalent of network switches. Enable JetStream, and they also become a new kind of DataBase server as well. 
+However, when it comes to ensuring the highest possible level of performance that NATS can provide it is good to keep a few things in mind.
+
+Think of Core NATS servers are a software equivalent of network switches. Enable JetStream, and they also become a new kind of DB server as well. 
 
 What you need to keep in mind is that when selecting the instance types and storage options for your NATS server host instances that in public clouds: *you get what you pay for*.
```
Comment (Jarema) [inline running-a-nats-service/environment.md:22]: added `keep in mind repetition`.

**Review by Jarema — COMMENTED** [review]
Looks good! Just a few comments 🙂

File: running-a-nats-service/environment.md [hunk 2]
```diff
@@ -11,24 +11,20 @@ If you do use load balancers you just need to understand the potential issues wi
...
 
-### Containerization
-
-#### Resource Limits
+### Resource Limits
 
 Be careful when setting resource limits for the nats-server containers. The nats-server processes use resources in proportion to the load traffic generated by all the client applications, if the NATS (and JetStream) usage is high (or bursty, nats-server is very fast and can process sharp bursts in traffic), then you will need to set the container resource limits accordingly, or the container orchestration system will kill the server's container. The nats-server - like all Golang executables - will see, and try to use, all host cores and all host memory, not the resource limits set for the container, and while you can set MAX_PROCS you can only influence max memory usage indirectly with GOGC.
```
[the remark this answers is not in the document]
Comment (jnmoyne) [reply running-a-nats-service/environment.md:30]: Good point, are to write a paragraph on `GOMEMLIMIT` and I'll add it? (or you can add a commit yourself too)

File: running-a-nats-service/environment.md (hunk 1 above)
Comment (jnmoyne) [reply running-a-nats-service/environment.md]: You mean you don't like the usage of 'keep in mind' twice? If so I changed the second one to a 'remember'

File: running-a-nats-service/environment.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jarema) [reply running-a-nats-service/environment.md]: I did commit those changes, as there were also changes around GOMAXPROCS in the server.

**Comment by gcolliso** [comment]
@jnmoyne is this one ready or do we still need additional review?

**Comment by jnmoyne** [comment]
Good to go, thanks!

**Review by gcolliso — APPROVED** [review]
lgtm

**Pull request merged**